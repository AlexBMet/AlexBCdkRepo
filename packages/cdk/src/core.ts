export const generateResourcePrefix = (uniquePrefix: string, environment: string, region: string) => {
	return `${uniquePrefix}-${environment}-${region}`
};
