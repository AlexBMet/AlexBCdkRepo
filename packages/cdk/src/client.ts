import { Bucket, HttpMethods } from '@aws-cdk/aws-s3';
import { CfnParameter, Construct, RemovalPolicy, Stack, StackProps, Tag } from '@aws-cdk/core';
import {Effect, PolicyStatement, ServicePrincipal} from "@aws-cdk/aws-iam";

export class Client extends Stack {
	public readonly deployBucketName: string;

	constructor(scope: Construct, id: string, props: StackProps) {
		super(scope, id, props);

		const STACK = Stack.of(this);
		const ENVIRONMENT = new CfnParameter(this, 'Environment', {
			allowedValues: ['dev', 'ci', 'stg', 'prod'],
			description: 'TBC',
			type: 'String',
		});
		const SERVICE_CODE = new CfnParameter(this, 'ServiceCode', {
			type: 'String',
			description: 'TBC',
		});
		const SERVICE_NAME = new CfnParameter(this, 'ServiceName', {
			type: 'String',
			description: 'TBC',
		});
		const SERVICE_OWNER = new CfnParameter(this, 'ServiceOwner', {
			type: 'String',
			description: 'TBC',
		});

		const BUCKET_NAME = new CfnParameter(this, 'BucketName', {
			type: 'String',
			description: 'TBC',
		});

		Tag.add(this, 'Environment', `${ENVIRONMENT.value}`);
		Tag.add(this, 'ServiceCode', `${SERVICE_CODE.value}`);
		Tag.add(this, 'ServiceName', `${SERVICE_NAME.value}`);
		Tag.add(this, 'ServiceOwner', `${SERVICE_OWNER.value}`);

		const websiteBucket = new Bucket(this, 'WebsiteBucket', {
			bucketName: `${BUCKET_NAME.value}`,
			removalPolicy: RemovalPolicy.DESTROY,
			websiteIndexDocument: 'index.html',
			websiteErrorDocument: 'error.html',
			publicReadAccess: true,
			cors: [
				{
					allowedOrigins: ['*'],
					allowedMethods: [HttpMethods.GET],
				},
			],
		});

		websiteBucket.addToResourcePolicy(
			new PolicyStatement({
				actions: ['s3:*'],
				effect: Effect.ALLOW,
				principals: [new ServicePrincipal('codebuild.amazonaws.com')],
				resources: [`${websiteBucket.bucketArn}/*`],
			})
		);

		this.deployBucketName = websiteBucket.bucketName;

		//new CfnOutput(this, 'websiteBucketName', { value: websiteBucket.bucketName })
	}
}
