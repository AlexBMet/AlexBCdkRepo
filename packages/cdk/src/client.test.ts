import { Client } from './client';
import { Stack } from '@aws-cdk/core';
import { SynthUtils } from '@aws-cdk/assert'; // tslint:disable-line:no-implicit-dependencies

describe('[client.ts] unit tests', () => {
	describe('[stack]', () => {
		it('must create an S3 bucket', async () => {
			const stack = new Client(new Stack(), 'ClientStack', {});
			expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
		});
	});
});
