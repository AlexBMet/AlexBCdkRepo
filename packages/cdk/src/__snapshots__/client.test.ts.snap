// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[client.ts] unit tests [stack] must create an S3 bucket 1`] = `
Object {
  "Parameters": Object {
    "BucketName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "Environment": Object {
      "AllowedValues": Array [
        "dev",
        "ci",
        "stg",
        "prod",
      ],
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceCode": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceName": Object {
      "Description": "TBC",
      "Type": "String",
    },
    "ServiceOwner": Object {
      "Description": "TBC",
      "Type": "String",
    },
  },
  "Resources": Object {
    "WebsiteBucket75C24D94": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": Object {
          "Ref": "BucketName",
        },
        "CorsConfiguration": Object {
          "CorsRules": Array [
            Object {
              "AllowedMethods": Array [
                "GET",
              ],
              "AllowedOrigins": Array [
                "*",
              ],
            },
          ],
        },
        "Tags": Array [
          Object {
            "Key": "Environment",
            "Value": Object {
              "Ref": "Environment",
            },
          },
          Object {
            "Key": "ServiceCode",
            "Value": Object {
              "Ref": "ServiceCode",
            },
          },
          Object {
            "Key": "ServiceName",
            "Value": Object {
              "Ref": "ServiceName",
            },
          },
          Object {
            "Key": "ServiceOwner",
            "Value": Object {
              "Ref": "ServiceOwner",
            },
          },
        ],
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketPolicyE10E3262": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "WebsiteBucket75C24D94",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": "*",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "WebsiteBucket75C24D94",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
